modules/rosa-hcp-sharing-vpc-dns-and-roles.adoc

```diff
@@ -0,0 +1,104 @@
+// Module included in the following assemblies:
+//
+// * networking/rosa-hcp-shared-vpc-config.adoc
+:_mod-docs-content-type: PROCEDURE
+[id="rosa-hcp-sharing-vpc-dns-and-roles_{context}"]
+= Step Two - Cluster Creator: Reserving your DNS and creating cluster Operator roles
+
+After the *VPC Owner* creates a virtual private cloud (VPC), subnets, and an IAM role for sharing the VPC resources, reserve an `openshiftapps.com` DNS domain and create Operator roles to communicate back to the *VPC Owner*.
+
+[NOTE]
+====
+For shared VPC clusters, you can choose to create the Operator roles after the cluster creation steps. The cluster will be in a `waiting` state until the Ingress Operator role ARN is added to the shared VPC role trusted relationships.
+====
+
+image::522-shared-vpc-step-2.png[]
+.Prerequisites
+
+* You have the `Route 53 role` ARN for the IAM role from the *VPC Owner*.
+* You have the `VPC endpoint role` ARN for the IAM role from the *VPC Owner*.
+
+.Procedure
+
+. Reserve an `openshiftapps.com` DNS domain with the following command:
++
+[source,terminal]
+----
+$ rosa create dns-domain --hosted-cp
+----
++
+The command creates a reserved `openshiftapps.com` DNS domain.
++
+[source,terminal]
+----
+I: DNS domain '14eo.p3.openshiftapps.com' has been created.
+I: To view all DNS domains, run 'rosa list dns-domains'
+----
+. Create an OIDC configuration.
++
+Review this article for more information on the link:https://access.redhat.com/articles/7031018[OIDC configuration process]. The following command produces the OIDC configuration ID that you need:
++
+[source,terminal]
+----
+$ rosa create oidc-config
+----
++
+You receive confirmation that the command created an OIDC configuration:
++
+[source,terminal]
+----
+I: To create Operator Roles for this OIDC Configuration, run the following command and remember to replace <user-defined> with a prefix of your choice:
+	rosa create operator-roles --prefix <user-defined> --oidc-config-id 25tu67hq45rto1am3slpf5lq6jargg
+----
+
+. Create the account roles by entering the following command:
++
+[source,terminal]
+----
+$ rosa create account-roles --oidc-config-id <oidc-config-ID> <1>
+    --installer-role-arn <Installer_Role> <2>
+    --route-53-role-arn <Created_Route_53_Role_Arn> <3>
```

Comment:
`route53-role-arn` (any instance of `route-53-role-arn` should have the leading `-` removed and become `route53` (unless we want to alias it, it has been made without the leading hyphen)